src/uwtools/tests/config/test_validator.py

```diff
@@ -135,13 +136,123 @@ def test_validate_workflow_tags_pass(vals):
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'command' is a required property")
+
+
+def test_validate_yaml_rocoto_invalid_no_task(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    del config_dict["workflow"]["tasks"]["metatask"]["task"]
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "{'var': {'member': 'foo bar baz'}} does not have enough properties")
+
+
+def test_validate_yaml_rocoto_invalid_no_var(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    del config_dict["workflow"]["tasks"]["metatask"]["var"]
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'var' is a required property")
+
+
+def test_validate_yaml_rocoto_invalid_type(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    config_dict["workflow"]["tasks"]["metatask"]["task"]["cores"] = "string"
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'string' is not of type 'integer'")
+
+
+def test_validate_yaml_rocoto_invalid_walltime_pattern(rocoto_assets, caplog):
+    schema_file, config_dict = rocoto_assets
+    config_dict["workflow"]["tasks"]["metatask"]["task"]["walltime"] = "0:01:00"
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
+        assert (
+            validator.validate_yaml(
+                schema_file=schema_file, config_file="/not/used", check_path=False
+            )
+            is False
+        )
+        assert logged(caplog, "'0:01:00' does not match '^[0-9]{2}:[0-9]{2}:[0-9]{2}$'")
+
+
+def test_validate_yaml_rocoto_valid(rocoto_assets):
+    schema_file, config_dict = rocoto_assets
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        yaml_config = YAMLConfig()
+        yaml_config.data = config_dict
```

Comment:
One more opportunity for brevity, if you're interested: The previous two lines can be simplified to
```
YAMLConfig().data = config_dict
```
That says: "The thing that the `Mock` object `YAMLConfig` returns when it is called -- i.e. the value of the expression `YAMLConfig()` -- should have an attribute called `data`, and that attribute's value should be `config_dict`".

My interest in making these tests concise is in making it as painless as possible to add more tests if we want them -- the less code not really accomplishing anything, the better, IMO.